awx/conf/signals.py

```diff
@@ -21,6 +22,14 @@ def handle_setting_change(key, for_delete=False):
     # When a setting changes or is deleted, remove its value from cache along
     # with any other settings that depend on it.
     setting_keys = [key]
+    # enabled/disable analytics sys job schedule with setting value
+    if key == 'INSIGHTS_TRACKING_STATE':
+        enabled = getattr(settings, key, None)
+        system_jt = SystemJobTemplate.objects.filter(name='Automation Insights Collection').first()
+        if system_jt is not None:
+            analytics_schedule = Schedule.objects.filter(name='Gather Automation Insights', unified_job_template=system_jt.id).first()
```

Comment:
If the schedule name changes, the enablement / disablement isn't going to work. I suggest getting all Schedule's related to the SystemJobTemplate and enable/disable via one Django `.update()` call.